code/modules/vehicles/mecha/_mecha.dm

```diff
@@ -357,18 +353,19 @@
 		. += "It's equipped with:"
 		for(var/obj/item/mecha_parts/mecha_equipment/ME in visible_equipment)
 			. += "[icon2html(ME, user)] \A [ME]."
-	if(!enclosed)
-		if(mecha_flags & SILICON_PILOT)
-			. += "[src] appears to be piloting itself..."
-		else
-			for(var/occupante in occupants)
-				. += "You can see [occupante] inside."
-			if(ishuman(user))
-				var/mob/living/carbon/human/H = user
-				for(var/O in H.held_items)
-					if(istype(O, /obj/item/gun))
-						. += "<span class='warning'>It looks like you can hit the pilot directly if you target the center or above.</span>"
-						break //in case user is holding two guns
+	if(enclosed)
+		return
+	if(mecha_flags & SILICON_PILOT)
+		. += "[src] appears to be piloting itself..."
+	else
+		for(var/occupante in occupants)
+			. += "You can see [occupante] inside."
+		if(ishuman(user))
+			var/mob/living/carbon/human/H = user
+			for(var/O in H.held_items)
+				if(istype(O, /obj/item/gun))
+					. += "<span class='warning'>It looks like you can hit the pilot directly if you target the center or above.</span>"
+					break //in case user is holding two guns
```

Comment:
```suggestion
			for(var/held_item in H.held_items)
				if(!isgun(held_item))
					continue
				. += "<span class='warning'>It looks like you can hit the pilot directly if you target the center or above.</span>"
				break //in case user is holding two guns
```